graphql_client_cli/src/generate.rs

```diff
@@ -25,8 +26,44 @@ pub fn generate_code(
         additional_derives,
         deprecation_strategy,
     };
+
     let gen = generate_module_token_stream(query_path, schema_path, Some(options))?;
-    let mut file = File::create(output)?;
-    write!(file, "{}", gen.to_string());
+
+    let mut file = File::create(output.clone())?;
+
+    let codes = gen.to_string();
+
+    if cfg!(feature = "rustfmt") && !no_formatting {
+        let codes = format(&codes);
+        write!(file, "{}", codes);
+        return Ok(());
+    }
+
+    write!(file, "{}", codes);
+
     Ok(())
 }
+
+#[allow(unused_variables)]
+fn format(codes: &str) -> String {
+    #[cfg(feature = "rustfmt")]
+    {
+        use rustfmt::{Config, Input, Session};
+        use std::default::Default;
+
+        let mut config = Config::default();
+
+        config.set().emit_mode(rustfmt_nightly::EmitMode::Stdout);
+        config.set().verbose(rustfmt_nightly::Verbosity::Quiet);
+
+        let mut out = Vec::with_capacity(codes.len() * 2);
+
+        Session::new(config, Some(&mut out))
+            .format(Input::Text(codes.to_string()))
+            .unwrap_or_else(|err| panic!("rustfmt error: {}", err));
+
+        return String::from_utf8(out).unwrap();
+    }
+    #[cfg(not(feature = "rustfmt"))]
+    codes.to_string()
```

Comment:
Since `format` should be called when the feature is disabled, we could even replace this with `unreachable!()`.